src/overlay/Overlay_D3D12.cpp

```diff
@@ -29,82 +28,182 @@ BOOL CALLBACK EnumWindowsProcMy(HWND hwnd, LPARAM lParam)
     return TRUE;
 }
 
-void Overlay::InitializeD3D12(IDXGISwapChain3* pSwapChain)
+bool Overlay::InitializeD3D12(IDXGISwapChain3* pSwapChain)
 {
-    EnumWindows(EnumWindowsProcMy, reinterpret_cast<LPARAM>(&m_hwnd));
+    static auto checkCmdQueue = [](Overlay* overlay) {
+        if (overlay->m_pCommandQueue == nullptr) 
+        {
+            auto swapChainAddr = reinterpret_cast<uintptr_t>(*(&overlay->m_pdxgiSwapChain));
+            overlay->m_pCommandQueue = *reinterpret_cast<ID3D12CommandQueue**>(swapChainAddr + kiero::getCommandQueueOffset());
+            if (overlay->m_pCommandQueue != nullptr) 
+            {
+                auto desc = overlay->m_pCommandQueue->GetDesc();
+                if(desc.Type != D3D12_COMMAND_LIST_TYPE_DIRECT) 
+                {
+                    overlay->m_pCommandQueue = nullptr;
+                    spdlog::warn("\tOverlay::InitializeD3D12(pSwapChain={0}) - invalid type of command list!", reinterpret_cast<void*>(*(&overlay->m_pdxgiSwapChain)));
+                    return false;
+                }
+                return true;
+            }
+            spdlog::warn("\tOverlay::InitializeD3D12(pSwapChain={0}) - swap chain is missing command queue!",  reinterpret_cast<void*>(*(&overlay->m_pdxgiSwapChain)));
+            return false;
+        }
+        return true;
+    };
+
+    static auto reset = [](Overlay* overlay) {
```

Comment:
line skip + same as above

Reply:
for now, corrected line skip here...

gonna leave it as lambda for now till you respond